app/models/katello/concerns/pulp_database_unit.rb

```diff
@@ -101,6 +129,13 @@ def import_for_repository(repository, force = false)
         self.sync_repository_associations(repository, :pulp_ids => ids) if self.manage_repository_association
       end
 
+      def import_for_pulp3_repository(repository, force = false)
+        ids = content_unit_class.ids_for_repository(repository.id)
+        ids_to_import = ids
+        self.pulp3_import_all(ids_to_import, :index_repository_association => false) if repository.content_view.default? || force
+        self.sync_repository_associations(repository, :pulp_ids => ids) if self.manage_repository_association
+      end
```

Comment:
one way to avoid these pulp3 vs pulp2 methods is to push them down to the service class (likely the pulp/repository.rb and pulp3/repository.rb service classes